src/lib/nodemailer.js

```diff
@@ -0,0 +1,71 @@
+const promisify = require("util").promisify;
+const nodemailer = require("nodemailer");
+const EMAIL_SIGNATURE = require("../services/emails/signature");
+
+class EmailService {
+
+    init({ email: user, clientId, clientSecret, refreshToken, accessToken }) {
+        this.email = user;
+        this.transporter = nodemailer.createTransport({
+            pool: true,
+            host: "smtp.gmail.com",
+            port: 465,
+            secure: true,
+            auth: {
+                type: "OAuth2",
+                user,
+                clientId,
+                clientSecret,
+                refreshToken,
+                accessToken,
+            },
+            connectionTimeout: 30000
+        });
+    }
+
+    verifyConnection() {
+        if (!this.transporter) throw new Error("Trying to use Email Service without having a transporter set. Call init() first.");
+        return promisify(this.transporter.verify)();
```

Comment:
Afaik `nodemailer` already does this internally?
> If `callback` argument is not set then the method returns a Promise object. Nodemailer itself does not use Promises internally but it wraps the return into a Promise for convenience.

In https://nodemailer.com/usage/#sending-mail

This is not extremely explicit in the "docs" though... They seem to be quite lacking...

Reply:
Yep, I knew about that (I am using it in the sendMail function), that one must have slipped...